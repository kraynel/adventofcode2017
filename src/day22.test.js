const { countInfectionBurst } = require("./day22");
const fs = require("fs");

const testMap = `..#
#..
...`;

describe("countInfectionBurst", () => {
  it("should countInfectionBurst", () => {
    expect(countInfectionBurst(testMap, 0)).toBe(0);
    expect(countInfectionBurst(testMap, 1)).toBe(1);
    expect(countInfectionBurst(testMap, 7)).toBe(5);
    expect(countInfectionBurst(testMap, 70)).toBe(41);
    expect(countInfectionBurst(testMap, 10000)).toBe(5587);
  });

  it("should get the value", done => {
    fs.readFile("inputs/day22.txt", "utf8", (err, data) => {
      expect(countInfectionBurst(data, 10000)).toMatchSnapshot();
      done();
    });
  });
});
