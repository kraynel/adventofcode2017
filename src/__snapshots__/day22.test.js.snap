// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countInfectionBurst should get the value 1`] = `5339`;
